ddtrace/_trace/tracer.py

```diff
@@ -826,7 +826,25 @@ def _start_span(
         dispatch("trace.span_start", (span,))
         return span
 
-    start_span = _start_span
+    def start_span(
```

Comment:
Context: `start_span = _start_span` was added when we wanted to remove the `start_span` function from our public api. We want all customers to use `tracer.trace(...)` instead. But we didn't around to deprecating and removing it.

Reply:
We should add a docstring to this method to make it clear that `start_span(...)`:
 - generates a root span by default  (child_of is None)
 - creates a span without activating it:

This block of code will not raise an assertion error:

```
s1 = tracer.start_span("root") 
assert tracer.current_span() == None
s2 = tracer.trace("root") 
assert s2._parent != s1
assert tracer.current_span() == s2
```

This behavior might surprise users.

Reply:
We should probably also discourage its use as well.